### DISCUSSION TITLE
Was kann das Ding?

### DISCUSSION THREAD

**Discussion by sabafreiburg** [hoylabs/OpenDTU-OnBattery#311] [Q&A]
Ich versuche noch zu verstehen was OpenDTU-OnBattery kann und was nicht. Sorry, das Readme gibt zwar viele, kleine Detailinfos, aber irgendwie bekomm ich keinen Kopf dran.
Skizze unten, was ich vor habe. Victron MPPTs sind keine Option, Batterieladung muß mit AC erfolgen. 
Für die Batterie Entladung wird der Batteriewechselrichter anhand der Werte vom Zähler so gesteuert, das keine Netzeinspeisung erfolgt.
Wie sieht das mit der Ladung, Ansteuerung des Huawei Netzteils aus? Um zu vermeiden, das mit Strom aus dem Netz geladen wird? Brauche ich da nicht noch den Status vom Solarwechselrichter, über die "normale" DTU, die da dran hängt?

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139558245/44c21065-506a-40ca-8d56-a01ff8ee61fa)

**Comment by madmartin** [comment]
Hallo @sabafreiburg 

das Bild was Du gemacht hast ist sehr gut.
Deine Frage zur Steuerung des Huawei Netzteils - da habe ich mal eine Hardcopy von dem entsprechenden Menü gemacht:

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/4208120/847fd951-b145-4100-b34e-ab9d81aaa2e0)

und soweit ich das verstanden habe (habe es selbst noch nicht in Betrieb) kann das Netzteil auch per MQTT gesteuert werden.

Wenn Du parallel zur Batterieanlage (untere Bildhälfte) ein "normales Balkonkraftwerk" betreibst (obere Bildhälfte) dann läuft das so: Das "Balkonkraftwerk" produziert Leistung und diese Leistung wird entweder intern verbraucht (und es wird entsprechend weniger aus dem Netz gezogen) oder sogar teilweise nach draussen gespeist (dann zeigt der Zähler Null Watt an).

In der OpenDTU-OnBattery wird die aktuelle Leistung aus dem Zähler (via Tasmota-Leser) alle paar Sekunden abgerufen und OpenDTU-OnBattery versucht dann die Leistungsabgabe den dort zugeordneten Wechselrichter in der unteren Bildhälfte  so zu steuern, daß der Stromzähler nur noch eine bestimmte Leistung (z.B. 20W) aus dem Netz zieht. 
Das funktioniert bei mir so gut, daß durch die "Unruhe" der Regelung nur noch 1-2 kWh pro Tag aus dem Netz gezogen werden, der Rest kommt dann aus Batterie (bei mir mit Victron Ladern) und dem "Balkonkraftwerk" (wie in Deiner Zeichnung).

**Reply by sabafreiburg** [reply]
Bs auf das Huawei Netzteil, Batterie, den zweiten Hoymiles und halt openDTU on Battery ist das bereits so in Betrieb.
Die Funktion mit Victron Ladern ist mir klar, wenn keine Leistung von den Panels kommt, wird die Batterie nicht mehr geladen. 
Das Huawei Netzteil hängt aber am AC Hausnetz und da kommt halt immer Strom. Ohne die Werte von dem ersten Hoymiles an den Panels, kann das m.E. nicht funktionieren.  Hab ich da was in der Doku übersehen?

**Reply by indie89** [reply]
Hi sabafreiburg, ich vermute, du hast da ein kleines Missverständnis: die aktuelle Leistung deiner PV ist völlig irrelevant, weil sie sozusagen am Zähler mit dem Bezug "verrechnet" (saldiert, wenn nicht auf der gleichen Phase) wird. Bedeutet, dass deine Optimierungsgröße nur der tasmota wert am Zähler ist. Der soll so nah an 0 ran, wie es geht. Weil du weder Solarstrom verschenken, noch nicht notwendigen Bezug haben möchtest. Der einspeisewert der PV ist da erstmal egal.
Mein Setup läuft genau wie oben in deinem Bild gezeigt, nur dass ich die Steuerung über einen iobroker mache, da ich bei voller Batterie meine Balkonkraftwerk(e) abregle, sodass maximal 80w eingespeist werden.

**Comment by Phantomias2006** [comment]
Hi,

ich baue gerade ähnliches. Bei mir erfolgt die Ladung auch ausschließlich mittels AC. 
Ich werde aber die Steuerung mittels NodeRed umsetzen. Also nutze ich openDTU on battery hauptsächlich um alles Informationen mittels MQTT zu übertragen und alles zu Steuern.

Ich lese direkt meinen Stromzähler aus. Hier sehe ich ob ich aktuell einen Bezug habe oder Einspeise. Nach diesem Wert werde ich regeln.

Der Speicher kommt aber erst gegen Ende nächster Woche. 
LG

**Reply by sabafreiburg** [reply]
Wenn du nach Stromzähler regelst, und auf jeden Fall vermeiden  möchtest aus dem  Netz zu laden, müsstest du als Sollwert IMMER eine geringe Einspeisung haben. Nur so weißt du, ob du Überschuss produzierst.
Zusätzlich noch NodeRed, nee... noch ein weiterer Prozess, der auf einem Rechner läuft,  der ausfallen kann. Der MQTT broker um den Stromzähler zu lesen, ist mir eigentlich schon zu viel.
Ein einziger ESP, der mit dem Solar- und dem Batteriewechselrichter kommuniziert und über CAN mit dem AC Lader, wäre mein Ziel.  Wenn das Ding ausfällt,  muss der Lader AC-seitig getrennt werden. Einen Tiefentladeschutz bekommt die Batterie sowieso.

**Reply by Phantomias2006** [reply]
Hi, der Stromzähler erkennt nicht nur den absoluten Zählerstand :-)
Anbei ein paar Werte welche ich zusätzlich auswerte:
![IMG_5776](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/c64f0294-41c0-4e0d-9134-1089e26e93fd)
1.7.0 ist Bezug und 2.7.0 die Produktion

**Comment by madmartin** [comment]
@sabafreiburg 
Einen Haken habe ich in der Zeichnung noch entdeckt:
Da ich annehme, daß Du das Laden der Batterie mit dem Netzteil irgendwie steuern willst (z.B. bei besonders günstigen Strom-Einkaufspreisen bei dynamischem Tarif) mußt Du dann während des Ladens den Dynamischen Power Limiter (DPL) abschalten.
Beim Ladevorgang wird ja ordentlich Energie aus dem Netz gezogen, was der DPL genau versucht auszugleichen und den Netzbezug auf den Sollwert nahe Null zu bringen.

Vielleicht hast Du Dir das aber doch ganz anders vorgestellt?
Deine Äußerung
>Ohne die Werte von dem ersten Hoymiles an den Panels, kann das m.E. nicht funktionieren. 

hat mich nämlich darauf gebracht  daß Du möglicherweise denkst, das "Balkonkraftwerk" produziert sagen wir mal 500W Leistung, und das Huawei-Netzeil soll nun genau diese 500W - abzüglich eines aktuellen Verbrauch im Haus - in die Batterie reinladen?
Dazu sage ich: das kann dieses Projekt so nicht. Das Netzteil ist dazu gedacht bei dynamischen Stromtarifen die Batterie mit "billigem Strom" aufzuladen.

Bitte positioniere dich mal klar, damit die Diskussion nicht in die falsche Richtung läuft.

**Reply by sabafreiburg** [reply]
Die Batterie soll in keinem Fall aus dem Netz geladen werden, darum ging es ja in meinem Eingangspost.
>>>>>
Wie sieht das mit der Ladung, Ansteuerung des Huawei Netzteils aus? Um zu vermeiden, das mit Strom aus dem Netz geladen wird? 
>>>>

**Reply by Phantomias2006** [reply]
@madmartin 
ist das sicher so? 
https://github.com/helgeerbe/OpenDTU-OnBattery/pull/288/commits/e279ce08c4ba84d363a732974cfbaf48e58bd812
Eigentlich dachte ich dass das mit diesem PR so funktioniert. Bei Überschuss wird mittels AC geladen.
LG

**Comment by sabafreiburg** [comment]
Ich schliesse die Frage hier erstmal. Habe die Hardware zusammen und werde mittels trial and error probieren.

**Comment by erwano1957** [comment]
@sabafreiburg :  Ich finde es schade, dass Du den Thread jetzt, wo es gerade spannend wird, schließen möchtest. (bzw. geschlossen hast)
Ich schlage vor, dass wir gemeinsam einen weiteren Anlauf versuchen. Folgende Bitte/Vorschlag möchte voranstellen.
Bitte lasst uns in der Fortführung jegliche Seitenthemen (Dynamische Tarife, DC-Kopplung etc.) vermeiden. Ansonsten besteht die Gefahr, dass wir vom ursprünglich von @sabafreiburg angesprochenen Thema entfernen, ohne es zu lösen. Schöne wäre auch, wenn hier möglichst eigene gemachte Erfahrungen und nicht Annahmen und Mutmaßungen einfließen würden.

Ich habe exakt die identische Anordnung vor, die @sabafreiburg eingangs sehr gut dargestellt hat. 
(**Erzeugung**: HM1500 + Hichi-Tasmota Lesekopf) + (**Nachteinspeisung**: HuaweiR488xx + HMxxx + Pylontech US2000C). 

Annahme: Nach meinem bisherigen Verständnis müsste die dargestellte Anordnung prinzipiell so funktionieren.

Folgende Fragen sind m.E. noch offen geblieben und ich würde mich freuen, wenn wir gemeinsam Antworten darauf finden könnten:
1. - Steuerung/Monitoring der Batterie: Hier finde ich widersprüchliche Angaben. Mal ist von CAN-Port die Rede, mal vom RS232 Consol-Port in Verbindung mit einer RS232-WLAN Umsetzung. Was ist hier sinnvoll?
2. - Huawei Charger: Hier holpere ich ebenfalls über die Kommunikationsschnittstelle. Welcher RS485- Baustein ist hier zu empfehlen. Idealerweise einer, der es nicht erfordert Leiterbahnen durchzutrennen, um sowohl 3,3V als auch 5 Volt verbinden zu können. Wenn aber eine Modifikation durchzuführen ist, wo ist das bitte beschrieben? 
(Bisher finde ich nur das von KlausI <BavarianSuperGuy> erstellte Video zu diesem Thema. Ist die erforderliche Modifikation, die er beschreibt, identisch mit der, die hier gebraucht wird?
3. - Zentrale Steuerung: Ich verwende Home Assistant und würde mich freuen, wenn die o.a. Konfiguration in HAS einzubinden wäre. Ist das via z.B. automatischer MQTT Erkennung möglich, oder muss ich dies manuell durchführen?
Ich bin gerne bereit, wenn wir dieses Thema gemeinsam lösen, anschließend eine umfassende Dokumentation für das Wiki zur Verfügung zu stellen, damit andere schneller zum Ziel kommen.
